In [2]:
import pandas as pd
import os
import pytablewriter
from io import StringIO

In [5]:
df_to_load = os.path.join("Resources","ch_district_summary_df.csv")
og_to_load = os.path.join("Resources", "district_summary_df.csv")
ch_ss_to_load = os.path.join("Resources", "ch_ss_df.csv" )
og_ss_to_load = os.path.join("Resources", "og_ss_df.csv" )
ch_df = pd.read_csv(df_to_load)
og_df = pd.read_csv(og_to_load)
ch_ss_df =pd.read_csv(ch_ss_to_load)
og_ss_df = pd.read_csv(og_ss_to_load)

In [20]:
ch_df

Total Schools Total Students    Total Budget  Average Math Score  \
0             15         39,170  $24,649,428.00           78.930533   

   Average Reading Score  % Passing Math  % Passing Reading  % Overall Passing  
0              81.855796       73.880521          84.651519          64.092418

In [21]:
og_df

Total Schools Total Students    Total Budget  Average Math Score  \
0             15         39,170  $24,649,428.00           78.985371   

   Average Reading Score  % Passing Math  % Passing Reading  % Overall Passing  
0               81.87784       74.980853          85.805463          65.172326

In [19]:
writer = pytablewriter.MarkdownTableWriter()
writer.table_name = "Corrected District DataFrame"
writer.header_list = list(ch_df.columns.values)
writer.value_matrix = ch_df.values.tolist()
writer.write_table()

# Corrected District DataFrame
|Total Schools|Total Students| Total Budget |Average Math Score|Average Reading Score|% Passing Math|% Passing Reading|% Overall Passing|
|------------:|--------------|--------------|-----------------:|--------------------:|-------------:|----------------:|----------------:|
|           15|39,170        |$24,649,428.00|             78.93|                81.86|         73.88|            84.65|            64.09|


 Corrected District DataFrame
|Total Schools|Total Students| Total Budget |Average Math Score|Average Reading Score|% Passing Math|% Passing Reading|% Overall Passing|
|------------:|--------------|--------------|-----------------:|--------------------:|-------------:|----------------:|----------------:|
|           15|39,170        |$24,649,428.00|             78.93|                81.86|         73.88|            84.65|            64.09|

In [24]:
writer = pytablewriter.MarkdownTableWriter()
writer.table_name = "Original District DataFrame"
writer.header_list = list(og_df.columns.values)
writer.value_matrix = og_df.values.tolist()
writer.write_table()

# Original District DataFrame
|Total Schools|Total Students| Total Budget |Average Math Score|Average Reading Score|% Passing Math|% Passing Reading|% Overall Passing|
|------------:|--------------|--------------|-----------------:|--------------------:|-------------:|----------------:|----------------:|
|           15|39,170        |$24,649,428.00|             78.99|                81.88|         74.98|            85.81|            65.17|


In [6]:
ch_ss_df

School Type  Total Students Total School Budget Per Student Budget  \
0     Charter            1635       $1,043,130.00            $638.00   

   Average Math Score  Average Reading Score  % Passing Math  \
0           83.350937              83.896082       66.911315   

   % Passing Reading  % Overall Passing Spending Ranges (Per Student)  \
0          69.663609          65.076453                      $630-644   

          School Size  
0  Medium (1000-2000)

In [7]:
og_ss_df

School Type  Total Students Total School Budget Per Student Budget  \
0     Charter            1635       $1,043,130.00            $638.00   

   Average Math Score  Average Reading Score  % Passing Math  \
0           83.418349               83.84893       93.272171   

   % Passing Reading  % Overall Passing Spending Ranges (Per Student)  \
0          97.308869          90.948012                      $630-644   

          School Size  
0  Medium (1000-2000)

In [11]:
writer = pytablewriter.MarkdownTableWriter()
writer.table_name = "Original Student Summary DataFrame"
writer.header_list = list(og_ss_df.columns.values)
writer.value_matrix = og_ss_df.values.tolist()
writer.write_table()

# Original Student Summary DataFrame
|School Type|Total Students|Total School Budget|Per Student Budget|Average Math Score|Average Reading Score|% Passing Math|% Passing Reading|% Overall Passing|Spending Ranges (Per Student)|   School Size    |
|-----------|-------------:|-------------------|------------------|-----------------:|--------------------:|-------------:|----------------:|----------------:|-----------------------------|------------------|
|Charter    |          1635|$1,043,130.00      |$638.00           |             83.42|                83.85|         93.27|            97.31|            90.95|$630-644                     |Medium (1000-2000)|


In [10]:
writer = pytablewriter.MarkdownTableWriter()
writer.table_name = "Modified School Summary DataFrame"
writer.header_list = list(ch_ss_df.columns.values)
writer.value_matrix = ch_ss_df.values.tolist()
writer.write_table()

# Modified School Summary DataFrame
|School Type|Total Students|Total School Budget|Per Student Budget|Average Math Score|Average Reading Score|% Passing Math|% Passing Reading|% Overall Passing|Spending Ranges (Per Student)|   School Size    |
|-----------|-------------:|-------------------|------------------|-----------------:|--------------------:|-------------:|----------------:|----------------:|-----------------------------|------------------|
|Charter    |          1635|$1,043,130.00      |$638.00           |             83.35|                83.90|         66.91|            69.66|            65.08|$630-644                     |Medium (1000-2000)|
